app/models/setting/content.rb

```diff
@@ -101,7 +107,18 @@ def self.load_defaults
                  nil, N_('Default Location Puppet content'), nil,
                  :collection => proc { Hash[Location.unscoped.all.map { |loc| [loc[:title], loc[:title]] }] }),
         self.set('expire_soon_days', N_('The number of days remaining in a subscription before you will be reminded about renewing it.'),
-                 120, N_('Expire soon days'))
+                 120, N_('Expire soon days')),
+        self.set('content_view_solve_dependencies',
+                 N_('Solve RPM dependencies on all Content View publishes. Publishes will take significantly longer with this option turned on ' \
+                 'and filters will be ignored to solve dependencies. Also, certain scenarios can still cause dependency errors. This can be ' \
+                 'set on individual Content Views as well.'),
+                 false, N_('Content View Dependency Solving')),
```

Comment:
I'm not sure i'm a fan of having this setting for a few reasons:
1) It only works in the UI (api/cli ignores it)
2) Its yet another configuration that duplicates an option on an individual content view
3) I think this is letting the use this when it doesn't make sense, since CVs without filters will still use this (needlessly)

Reply:
The suggestion of a global setting came from @sideangleside so I'll let him respond

I think having a better idea of how publish time is affected will help us make this decision too, I'll do some rough benchmarking soon

Reply:
A few thoughts that are fresh in my brain so I'll share them :)

> I think this is letting the user use dep solving when it doesn't make sense, since CVs without filters will still use this (needlessly)

It would be nice to add an option to not dependency solve if there are no filters (as we discussed on IRC)

> I think this is letting the user use dep solving when it doesn't make sense, since CVs without filters will still use this (needlessly)

This is a fair point, the setting combined with the 'skip dep solving' option does complicate things. I'll think of some better ways to go about this

Reply:
I want to 'idiot proof' the setup as much as possible. If the performance hit is small-ish, I can see users setting this globally (as the ends justify the means). However, if the performance hit is significant, I would be ok not having a global setting.